week-4-practise/level-2/exercise-1.js

```diff
@@ -9,17 +9,19 @@ We made a start on this for you here:
 */
 
 var bottle = {
-	volume: 0,
-	fill: function(){
-	 	// calling this function should make you bottles volume = 100; 
-	},
-	drink: function(){
-		// calling this function should decrease your bottles volume by 10;
-	},
-	empty: function(){
-		// this function should return true if your bottles volume = 0
-	}
-}
+  volume: 0,
+  fill: function() {
+    return (this.volume = 100);
+  },
+  drink: function() {
+    return (this.volume = this.volume - 10);
+  },
+  empty: function() {
+    if (this.volume <= 0) {
+      return true;
+    }
+  }

```

Comment:
There's a small bug here, waiting to bite you.  

What will your code return if the volume is greater than zero?  

When you see `if (condition) { return }` but no `else`, this should ring alarm bells in your head because there's a path through your code which doesn't explicitly return anything.  It's not *always* wrong, but mostly.

Reply:
Thanks for pointing it out!!